src/native/external/zlib-intel/deflate.c

```diff
@@ -1553,10 +1553,10 @@ local block_state deflate_stored(s, flush)
         _tr_stored_block(s, (char *)0, 0L, last);
 
         /* Replace the lengths in the dummy stored block with len. */
-        s->pending_buf[s->pending - 4] = len;
-        s->pending_buf[s->pending - 3] = len >> 8;
-        s->pending_buf[s->pending - 2] = ~len;
-        s->pending_buf[s->pending - 1] = ~len >> 8;
+        s->pending_buf[s->pending - 4] = (Bytef)len;
```

Comment:
Here and elsewhere: I think these added castings need a comment above them explaining the reason of the castings. Especially since this is diverging from the original code.

Reply:
I considered this but opted against it because it doesn't fit the style normally used by the project maintainer (see https://github.com/madler/zlib/commit/3df842426b53522e232da7aa06d5ef10eeb5ec4a). Instead, I tried to match his style as closely as possible - including not putting source comments - and put all the comments in the .patch file itself. That should hopefully make it easier to send this upstream as a proper patch and will minimize any divergence between our sources and the official sources.

Reply:
Thanks for the explanation.